src/core/ImperativeBase.ts

```diff
@@ -182,6 +182,8 @@ export class ImperativeBase extends Settable(Transformable) {
 		// Helpful for debugging when looking in devtools.
 		// @prod-prune
 		o.name = `${this.tagName}${this.id ? '#' + this.id : ''} (webgl, ${o.type})`
+		// Save LUME element ref to Three.js object
+		o.userData.lumeElement = this
```

Comment:
For this we can use a Map so that it is typesafe. We'll need to eventually once we compile to WebAssembly anyway. We also need to unmap the pair on disconnect.

Reply:
Yeah I think just a weak map would be good

Reply:
Another idea is we can make a subclass mix in or factory function that returns an instance with the lumeElement property

Reply:
Would the map be stored on the scene to keep track of each `Object3d`'s corresponding Lume element? Then when it's time to dispatch the event, the scene gets the object's element and dispatches it that way?

Reply:
It can just be in the top level of a module. We can export a function so that anyone can use it, for example:

```js
import {getLumeElementFromThree} from 'lume'

getLumeElementFromThree(someThreejsObject)
```

and what this might do is traverse up until it finds a node associated with a Lume element using that map.